app/stac_api/serializers.py

```diff
@@ -329,6 +329,7 @@ class Meta:
     summaries = serializers.JSONField(read_only=True)
     stac_extensions = serializers.SerializerMethodField(read_only=True)
     stac_version = serializers.SerializerMethodField(read_only=True)
+    title = serializers.CharField(allow_blank=False, allow_null=False, default=None)
```

Comment:
This is to avoid empty titles to be shown, I guess.
I am not sure, if the `allow_blank=False` and `allow_null=False` are needed here.
But I guess, a `required=False` could make sense here. Perhaps additionally or to replace the `allow_blank` and `allow_null` options.

http://www.tomchristie.com/rest-framework-2-docs/api-guide/serializers
"ote: When validation is applied to a ModelSerializer, both the serializer fields, and their corresponding model fields must correctly validate. If you have optional fields on your model, make sure to correctly set blank=True on the model field, as well as setting required=False on the serializer field."

Reply:
@hansmannj This change has been made that the endpoint only returns a title attribute, when it has content

Reply:
Good point we must set `required=False` in the serializer for optional fields. @hansmannj having here `required=False`, `allow_blank=False` (or `allow_null=False` depending on the field type) and `default=None` allow the framework to have the field optional and to have the PUT method behave correctly, meaning that if the field is not in the PUT request, it is set to None in the DB, which removes the field from the answer due to our `NonNullModelSerializer` base class. Whithout this config, the PUT request will behave like a PATCH in case an optional field is missing in the request body.
That being said to be consistent with the Item serializer this title field should be:
`title = serializers.CharField(required=False, allow_blank=False, default=None, max_length=255)`
It may be good to have a comment about this in code (adding it also for the Item.properties.title), this is for the moment only commented in git commit https://github.com/geoadmin/service-stac/pull/134/commits/c07541989c19a1cf348f1737ccaa48d431399096

I would remove the `allow_null=False` because it is anyway the default value and because of this comment from the Rest Framework docu (https://www.django-rest-framework.org/api-guide/fields/#charfield)

> The allow_null option is also available for string fields, although its usage is discouraged in favor of allow_blank. It is valid to set both allow_blank=True and allow_null=True, but doing so means that there will be two differing types of empty value permissible for string representations, which can lead to data inconsistencies and subtle application bugs.